src/main/kotlin/org/rust/ide/inspections/ApproxConstantInspection.kt

```diff
@@ -3,39 +3,28 @@ package org.rust.ide.inspections
 import com.intellij.codeInspection.ProblemsHolder
 import com.intellij.psi.PsiElement
 import com.intellij.psi.PsiElementVisitor
-import org.rust.lang.core.psi.RustLitExpr
-import org.rust.lang.core.psi.RustVisitor
+import org.rust.lang.core.psi.RustAttr
+import org.rust.lang.core.psi.RustLiteral
+import org.rust.lang.core.psi.util.parentOfType
+import org.rust.lang.core.psi.visitors.RustVisitorEx
 
 class ApproxConstantInspection : RustLocalInspectionTool() {
-
     override fun getDisplayName() = "Approximate Constants"
 
-    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
-        return object : RustVisitor() {
-            override fun visitLitExpr(o: RustLitExpr) {
-                analyzeLiteral(o.floatLiteral ?: return, holder)
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor =
+        object : RustVisitorEx() {
+            override fun visitNumericLiteral(literal: RustLiteral.Number) {
+                if (literal.isFloat && literal.parentOfType<RustAttr>() == null) {
```

Comment:
I worry about this `literal.parentOfType<RustAttr> == null` It will walk the tree up to the top, so potentially this is `O(n^2)` to annotate the whole file. I think it's better to remove it, or to add the opposite check like `literal.parentOfType<Expression> != null`.